jujugui/static/gui/src/app/components/deployment-flow/cloud/cloud.js

```diff
@@ -37,28 +40,37 @@ YUI.add('deployment-cloud', function() {
     },
 
     componentWillMount: function() {
-      this.props.listClouds((error, clouds) => {
-        if (error) {
-          console.error('Unable to list clouds', error);
+      const listClouds = () => {
+        // It is possible that the controller hasn't yet connected so
+        // bounce until it's connected then fetch the clouds list.
+        if (!this.props.controllerIsAvailable()) {
+          setTimeout(listClouds, 500);
```

Comment:
if the controller is not available it will call this function `listClouds` again, which will check if the controller is available and repeat.